packages/core/schematics/migrations/static-queries/index.ts

```diff
@@ -22,34 +23,82 @@ import {TimingStrategy} from './strategies/timing-strategy';
 import {QueryUsageStrategy} from './strategies/usage_strategy/usage_strategy';
 import {getTransformedQueryCallExpr} from './transform';
 
-type Logger = logging.LoggerApi;
-
 /** Entry point for the V8 static-query migration. */
 export default function(): Rule {
   return (tree: Tree, context: SchematicContext) => {
-    const projectTsConfigPaths = getProjectTsConfigPaths(tree);
-    const basePath = process.cwd();
-
-    if (!projectTsConfigPaths.length) {
-      throw new SchematicsException(
-          'Could not find any tsconfig file. Cannot migrate queries ' +
-          'to explicit timing.');
-    }
-
-    for (const tsconfigPath of projectTsConfigPaths) {
-      runStaticQueryMigration(tree, tsconfigPath, basePath, context.logger);
-    }
+    // We need to cast the returned "Observable" to "any" as there is a
+    // RxJS version mismatch that breaks the TS compilation.
+    return from(runMigration(tree, context).then(() => tree)) as any;
   };
 }
 
+/** Runs the V8 migration static-query migration for all determined TypeScript projects. */
+async function runMigration(tree: Tree, context: SchematicContext) {
+  const projectTsConfigPaths = getProjectTsConfigPaths(tree);
+  const basePath = process.cwd();
+  const logger = context.logger;
+
+  logger.info('------ Static Query migration ------');
+  logger.info('In preparation for Ivy, developers can now explicitly specify the');
+  logger.info('timing of their queries. Read more about this here:');
+  logger.info('https://github.com/angular/angular/pull/28810');
+  logger.info('');
+
+  if (!projectTsConfigPaths.length) {
+    throw new SchematicsException(
+        'Could not find any tsconfig file. Cannot migrate queries ' +
+        'to explicit timing.');
+  }
+
+  // In case prompts are supported, determine the desired migration strategy
+  // by creating a choice prompt. By default the template strategy is used.
+  let isUsageStrategy = false;
+  if (supportsPrompt()) {
+    logger.info('There are two available migration strategies that can be selected:');
+    logger.info('  • Template strategy  -  (safe migration, uses anti-patterns)');
+    logger.info('  • Usage strategy  -  (uses best practices, not always correct)');
+    logger.info('For an easy migration, the template strategy is recommended. The usage');
+    logger.info('strategy can be used for best practices and easier migration to Ivy.');
```

Comment:
Template strategy - migration tool (short-term gains, rare corrections)
Usage strategy - best practices (long-term gains, manual corrections)

```suggestion
    logger.info('strategy can be used for best practices and a code base that will be more flexible to changes going forward.');
```

Reply:
Sounds good to me. done!